spec/frodo/profile_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+describe Frodo::Profile do
+  let(:location) do
+    OpenStruct.new(id: 'asdf')
+  end
+
+  let(:token) { 'Foobarbaz' }
+
+  let(:headers) do
+    { Authorization: "Bearer #{token}" }
+  end
+
+  before do
+    ENV['GANDALF_URL'] = 'https://www.gandalf.com'
+    stub_request(:get, ENV['GANDALF_URL'] + '/location/asdf/profile')
+      .with(headers: headers)
+      .to_return(status: status, body: body, headers: {})
+  end
+
+  context '#profile' do
+    subject { described_class.instance(location: location, token: token).data }
+
+    context 'errors' do
+      context "when there's a JSON parsing error" do
+        let(:status) { 200 }
+        let(:body) { 'stupidbody: , bleh' }
+
+        it 'returns a JsonError' do
+          expect { subject }.to raise_error(Frodo::Errors::JsonError)
+        end
+      end
+
+      context 'all other errors' do
+        let(:status) { 401 }
+        let(:body) do
+          JSON.unparse(error: 'The access token expired')
+        end
+
+        it 'returns an ProfileError' do
+          expect { subject }.to raise_error(Frodo::Errors::ProfileError)
+        end
+      end
+    end
+
+    context 'with valid params' do
+      let(:status) { 200 }
+
+      let(:body) do
+        JSON.unparse(id: 'dfb70c18-0677-4e4f-bce4-17158eba0ecf',
```

Comment:
Wait this is not the response from the endpoint, it goes like this:

```
            "data": {
                "id": "33564fdb-2366-45e7-8d90-ddb85d16f9d0",
                "type": "profiles",
                "attributes": {
                    "version": 1,
                    "data": {
                        "configs": 
```

Reply:
wait... data->attributes->data->configs ?

i think we should get rid of the second `data` maybe ?